spec/factories/messages.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: messages
+#
+#  id               :bigint           not null, primary key
+#  messageable_type :string           not null
+#  text             :text             not null
+#  created_at       :datetime         not null
+#  updated_at       :datetime         not null
+#  messageable_id   :integer          not null
+#  user_id          :integer          not null
+#
+FactoryBot.define do
+  factory :message do
+    association :user
+    text { Faker::Lorem.paragraph }
+    messageable_id { user_id }
+    messageable_type { 'user' }
+
+    trait :without_user do
+      user_id { nil }
```

Comment:
I'm not sure, but I think that should be possible. Please ignore if it leads to an error.
```suggestion
      user { nil }
```

Reply:
You may also want to reconsider having `:without_user` trait on factory - this is an invalid state as user's presence is validated, so there is no sense creating traits for building invalid records.